lib/string/README

```diff
@@ -0,0 +1,270 @@
+General guidelines:
+===================
+
+-  If there's an upper-case macro that wraps a function, use the macro
+   if possible.  These use macro magic to add safety.
+
+-  x*() functions wrap a function of the same name without the 'x'.
+   These wrappers exit on error instead of returning NULL.
+   They simplify program code.  Avoid them in library code.
+
+-  strn*() functions are forbidden for use with strings.  Consider strn
+   a mnemonic for [[gnu::nonstring]].  We use them with utmp(5) members
+   (almost) exclusively.
+
+
+Don't use some libc functions without Really Good Reasons:
+==========================================================
+
+<stdio.h>
+    asprintf(3)
+	Use aprintf() instead.
+	It is difficult to handle errors after asprintf(3).  Also, it
+	makes it more difficult for static analyzers to check that calls
+	to it free(3) memory correctly.
```

Comment:
s/to it free(3)/to free(3)

Reply:
> In that case we should hide this by making the function static in the same file as stprspn()

stpspn() can only be implemented as a macro, so it's in a header file, which means we need strrspn() to be an extern function.

Reply:
> s/to it free(3)/to free(3)

I agree this reads weirdly, but the `it` is correct there.  I'll reword to avoid this weirdness.